### PR TITLE
Make pipeline handle moving main results

### PR DESCRIPTION
**Pull request by charleskawczynski**
Refs:
 - https://buildkite.com/docs/apis/rest-api/artifacts
 - https://buildkite.com/docs/pipelines/dependencies#main
 - https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars

### CODE REVIEW

**Comment by jakebolewski** [comment]
LGTM

**Review by jakebolewski — APPROVED** [review]

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -167,8 +167,16 @@ steps:
 
   - label: ":robot_face: Move main results"
     key: "cpu_mv_main_res"
+    # if: build.branch == "staging" # add back in after debugging
     command:
-      - "julia --color=yes --project utils/move_output.jl"
+      - commit_sha="${BUILDKITE_COMMIT}" | cut -c1-7
+      - cluster_data_prefix="/central/scratch/climaci/turbulenceconvection-main/${commit_sha}/"
+      - tmp_dir=$(mktemp -d)
+      - echo commit_sha
```
Comment (jakebolewski) [inline .buildkite/pipeline.yml:175]: these need to be interpolated $COMMIT_SHA

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -167,8 +167,16 @@ steps:
...
+      - echo cluster_data_prefix
+      - mv Output.* ${tmp_dir}
+      - mv ${tmp_dir} ${cluster_data_prefix}
```
Comment (jakebolewski) [inline .buildkite/pipeline.yml:178]: you should also always quote interplated paths so they escape properly:

mv "${tmp_dir}" "${cluster_data_prefix}"

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -167,8 +167,16 @@ steps:
 
   - label: ":robot_face: Move main results"
     key: "cpu_mv_main_res"
+    # if: build.branch == "staging" # add back in after debugging
     command:
-      - "julia --color=yes --project utils/move_output.jl"
+      - commit_sha="${BUILDKITE_COMMIT}" | cut -c1-7
+      - cluster_data_prefix="/central/scratch/climaci/turbulenceconvection-main/${commit_sha}/"
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply .buildkite/pipeline.yml:173]: Ah, crud, that would be really convenient. Can we easily change that?

[the remark this answers is not in the document]
Comment (jakebolewski) [reply .buildkite/pipeline.yml:173]: A path could be `${BUILDKITE_BUILD_PATH}/${BUILDKITE_PIPELINE_SLUG}/main-results/${BUILDKITE_COMMIT}`  that way you don't even need to process the short sha and would "just work" when re-configuring the runners for any pipeline.

**Comment by charleskawczynski** [comment]
@jakebolewski I tried following [this comment](https://github.com/CliMA/TurbulenceConvection.jl/pull/114#discussion_r687894717), but I don't think I did it right 🙁. See anything out of place?

**Comment by jakebolewski** [comment]
you are not escaping the interpolation correctly (there is before an agent run, "build time" interpolation and "runtime interpolation), it's kind of weird but common with yaml configs here are the rules: 

https://buildkite.com/docs/agent/v3/cli-pipeline#environment-variable-substitution

so you want $$VAR instead of $VAR for runtime interpolation

**Comment by jakebolewski** [comment]
also you still want to keep the commit hash, it should be `$$PREFIX/$$SHA` otherwise you'll lose consistency when the results are written but the staged commit not yet merged into `main`.

the state sequence is roughly:
* PR run triggered, reads results from `main` at commit `PREV_SHA`, path `PREFIX/PREV_SHA`
* staged PR running, staged results copied into `PREFIX/SHA`
* bors updates branch `main` to point to staged commit `SHA`
* concurrent PR runs during this process still read from `PREFIX/PREV_SHA` (maintaining consistent view when they were triggered)
* new PR runs will read from `PREFIX/SHA`

**Pull request closed without merging**